### PR TITLE
add net.sourceforge.kmetronome

### PR DESCRIPTION
**Pull request by pedrolcl**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(This software uses hardware MIDI devices, and may need user files.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link: https://sourceforge.net/projects/kmetronome/**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by pedrolcl** [comment]
bot, build

**Comment by pedrolcl** [comment]
bot, build net.sourceforge.kmetronome

**Comment by pedrolcl** [comment]
bot, build net.sourceforge.kmetronome

**Comment by pedrolcl** [comment]
bot, build net.sourceforge.kmetronome

File: net.sourceforge.kmetronome.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://downloads.sourceforge.net/project/drumstick/2.5.0/drumstick-2.5.0.tar.bz2",
+          "sha256": "8709d8a7c438c1d09091827ce5fd8286d387a046a8f717a1a34546d9ca3d902a"
+        },
+        {
+          "type": "patch",
+          "path": "install-header-macros.patch"
```
Comment (hfiguiere) [inline net.sourceforge.kmetronome.json:59]: this patch is missing.

File: net.sourceforge.kmetronome.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  "command": "kmetronome",
+  "rename-icon": "kmetronome",
+  "finish-args": [
+    /* X11 + XShm access */
+    "--share=ipc",
+    "--socket=x11",
+    /* MIDI */
+    "--device=all",
+    /* Allow loading, saving files from anywhere (portals don’t work yet) */
+    "--filesystem=home",
```
Comment (hfiguiere) [inline net.sourceforge.kmetronome.json:15]: does a metronome really need to access the whole home directory?

File: net.sourceforge.kmetronome.json [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+          "path": "install-header-macros.patch"
+        }
+      ]
+    },
+    {
+      "name": "kmetronome",
+      "buildsystem": "cmake-ninja",
+      "config-opts": [ "-DBUILD_DOCS=OFF" ],
+      "post-install": [
+        "install -Dm644 -t /app/share/metainfo/ net.sourceforge.kmetronome.appdata.xml"
```
Comment (hfiguiere) [inline net.sourceforge.kmetronome.json:68]: This is not installed by upstream? (it should be in a future release)

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
also the `launchable` in upstream appdata file is wrong since it doesn't match the name of the `.desktop`

File: net.sourceforge.kmetronome.json (hunk 2 above)
Comment (pedrolcl) [reply net.sourceforge.kmetronome.json:15]: The user may create pattern files any where, and they should be reloaded automatically upon startup (i.e. not only interactive loaded on demand)

File: net.sourceforge.kmetronome.json (hunk 1 above)
Comment (pedrolcl) [reply net.sourceforge.kmetronome.json:59]: Thanks. Added now

File: net.sourceforge.kmetronome.json (hunk 3 above)
Comment (pedrolcl) [reply net.sourceforge.kmetronome.json:68]: Yes, a small mistake on my part. It is included in the SVN now: 
https://sourceforge.net/p/drumstick/code/623/

**Comment by pedrolcl** [comment]
bot, build net.sourceforge.kmetronome

File: net.sourceforge.kmetronome.json (hunk 3 above)
Comment (pedrolcl) [reply net.sourceforge.kmetronome.json:69]: I'm following here the same post-install step of VMPK and dmidiplayer. Do you mean that it is wrong? Or that I should remove this step entirely?

**Comment by pedrolcl** [comment]
bot, build net.sourceforge.kmetronome

**Pull request closed without merging**